sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -44,6 +44,9 @@ class StringType private (val collationId: Int) extends AtomicType with Serializ
   private[sql] def supportsLowercaseEquality: Boolean =
     CollationFactory.fetchCollation(collationId).supportsLowercaseEquality
 
+  private[sql] def isNonCSAICollation: Boolean =
+    !CollationFactory.isAI(collationId) || CollationFactory.isCI(collationId)
```

Comment:
this is starting to look very confusing at this point - I would avoid adding this to `StringType`